File: src/types/JobEvent.ts
export interface JobEvent {
  cmd_name: string;
  resource_path: string;
  created_at: string;
  ui_action?: UIAction;
}

export enum UIAction {
  PasswordRequired = "PasswordRequired",
}


export const getJobId = (job: JobEvent) => `${job.cmd_name}@${job.resource_path}`;

File: src-tauri/src/jobs/job_event.rs
use serde::Serialize;

#[derive(Clone, Serialize)]
// #[serde(rename_all = "snake_case")]
pub enum UIAction {
    PasswordRequired
}

#[derive(Clone, Serialize)]
pub struct JobEvent {
    pub cmd_name: String,
    pub resource_path: String,
    pub created_at: String,
    pub ui_action: Option<UIAction>,
}
